plugin.audio.ampache/default.py

```diff
@@ -0,0 +1,1070 @@
+    image_url = check_get_art_url(node)
+    try:
+        album_id = getSuperId(node,"album")
+        albumArt = art.get_art(album_id,"album",image_url)
+    except:
+        albumArt = art.get_art(None,"album",image_url)
+    liz.setLabel(str(node.findtext("title")))
+    liz.setArt( art.get_artLabels(albumArt) )
+    #needed by play_track to play the song, added here to uniform api
+    liz.setPath(node.findtext("url"))
+    liz.setInfo( type="music", infoLabels=get_infolabels("songs", node) )
+    liz.setMimeType(node.findtext("mime"))
+
+def addLinks(elem,object_type,useCacheArt,mode):
+
+    image = "DefaultFolder.png"
+    elem_type = ut.otype_to_type(object_type)
+    it=[]
+    allid = set()
+
+    for node in elem.iter(elem_type):
+        cm = []
+        object_id = int(node.attrib["id"])
+        #xbmc.log("AmpachePlugin::addLinks: object_id  - " + str(object_id) , xbmc.LOGDEBUG )
+        #xbmc.log("AmpachePlugin::addLinks: node " + ET.tostring(node) , xbmc.LOGDEBUG )
+
+        name = str(node.findtext("name"))
+
+        if elem_type == "album":
+            try:
+                #no unicode function, cause urllib quot_plus error ( bug )
+                #album_id is ok also as string, cause it is needed to create
+                #an url
+                #remove duplicates in album names ( workaround for a problem in server comunication )
+                if object_id not in allid:
+                    allid.add(object_id)
+                else:
+                    continue
+                try:
+                    artist_id = getSuperId(node,"artist")
+                    cm.append( ( ut.tString(30141), "Container.Update(%s?object_id=%s&mode=2)" % ( sys.argv[0],artist_id ) ) )
+                except:
+                    pass
+
+                name = get_album_artist_name(node)
+                if useCacheArt:
+                    image_url = check_get_art_url(node)
+                    image = art.get_art(object_id,elem_type,image_url)
+            except:
+                xbmc.log("AmpachePlugin::addLinks: album_id error", xbmc.LOGDEBUG)
+        else:
+            useCacheArt = False
+
+        infoLabels=get_infolabels(object_type,node)
+
+        if infoLabels == None:
+            infoLabels={ "Title": name }
+
+        liz=xbmcgui.ListItem(name)
+        liz.setInfo( type="Music", infoLabels=infoLabels )
+
+        if useCacheArt:
+            #faster loading for libraries
+            liz.setArt(  art.get_artLabels(image) )
+        liz.setProperty('IsPlayable', 'false')
+
+        if cm:
+            liz.addContextMenuItems(cm)
+
+        handle=int(sys.argv[1])
+
+        u=sys.argv[0]+"?object_id="+str(object_id)+"&mode="+str(mode)+"&name="+urllib.parse.quote_plus(name)
+        #xbmc.log("AmpachePlugin::addLinks: u - " + u, xbmc.LOGDEBUG )
+        isFolder=True
+        tu= (u,liz,isFolder)
+        it.append(tu)
+
+    ok=xbmcplugin.addDirectoryItems(handle=int(sys.argv[1]),items=it,totalItems=len(elem))
```

Comment:
returning here doesn't seem to be necessary

Reply:
I will remove

Reply:
done